### PR TITLE
vim-patch:7.4.1140

### PR DESCRIPTION
**Pull request by prollings**
Problem:    Recognizing <sid> does not work when the language is Turkish.
            (Christian Brabandt)
Solution:   Use MB_STNICMP() instead of STNICMP().

Note: MB_STRNICMP has been renamed mb_strnicmp
Note2: Added a test for this

https://github.com/vim/vim/commit/e266d6d664d6d743c79797af400b2c01ec746216

### CODE REVIEW

**Comment by KillTheMule** [comment]
There's a bug description [here](https://groups.google.com/forum/#!topic/vim_dev/dGrHd0YMgWI). I wonder if one can make a test for that?

**Comment by KillTheMule** [comment]
Ok I investigated this a bit, with help from gitter. You can simply set the locale from nvim using `:lang ctype tr-TR.utf-8` (which might throw an error, in which case the test should be marked as pending, see `test/functional/terminal/buffer_spec.lua`) and then trigger the error by using

```
source([[
set nocp
func! <sid>_dummy_function()
        echo 1
endfunc
au VimEnter * call <sid>_dummy_function() 
]])
```

File: src/nvim/eval.c [hunk 1]
```diff
@@ -20155,11 +20155,15 @@ trans_function_name (
  */
 static int eval_fname_script(char_u *p)
 {
-  if (p[0] == '<' && (STRNICMP(p + 1, "SID>", 4) == 0
-                      || STRNICMP(p + 1, "SNR>", 4) == 0))
+  // Use mb_stricmp() because in Turkish comparing the "I" may not work with
+  // the standard library function. */
```
Comment (jamessan) [inline src/nvim/eval.c]: Drop the trailing ` */` here.

File: test/functional/viml/lang_spec.lua [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+local clear, eval, eq = helpers.clear, helpers.eval, helpers.eq
+local execute, source = helpers.execute, helpers.source
+
+describe('turkish', function()
+  before_each(clear)
+
+  it('applies locale to \'i\' in `<SID>` comparison', function()
+    execute('lang ctype tr_TR.UTF-8')
+    if string.find(eval('v:errmsg'), '^E197: ') then
+      pending("Locale tr_TR.UTF-8 not supported")
```
Comment (jamessan) [inline test/functional/viml/lang_spec.lua]: Should have a `return` after the pending line so we don't execute the rest of the test.

File: test/functional/viml/lang_spec.lua [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      return
+    end
+    source([[
+      set nocp
```
Comment (oni-link) [inline test/functional/viml/lang_spec.lua]: Do we need `set nocp`, is it not always set?

**Comment by justinmk** [comment]
Verified test. Nice work.

**Pull request merged**